### PR TITLE
Add method for searching bookings for particular Hotel with particular timing

### PR DESCRIPTION
**Pull request by blackberts**

### CODE REVIEW

**Review by Genresal — APPROVED** [review]

File: BKG/Booking.API/Controllers/HotelController.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 using Booking.API.ViewModels;
 using Booking.BLL.Models;
 using Booking.BLL.Services.Generic;
+using Microsoft.AspNetCore.Mvc;
```
Comment (nantonov) [inline BKG/Booking.API/Controllers/HotelController.cs]: Unnecessary using?

File: BKG/Booking.BLL/DI/DependencyInjection.cs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ﻿using Booking.BLL.Mapping;
 using Booking.BLL.Models;
+using Booking.BLL.Services.Booking;
 using Booking.BLL.Services.Generic;
+using Booking.BLL.Services.Booking;
```
Comment (nantonov) [inline BKG/Booking.BLL/DI/DependencyInjection.cs]: Unnecessary using?

**Review by nantonov — CHANGES_REQUESTED** [review]

File: BKG/Booking.BLL/Services/Booking/IBookingService.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using Booking.BLL.Models;
+using Booking.BLL.Services.Generic;
+
+namespace Booking.BLL.Services.Booking
```
Comment (CrazyKamenwik) [inline BKG/Booking.BLL/Services/Booking/IBookingService.cs]: Put interfaces into separate folder like "Interfaces"

File: BKG/Booking.BLL/Services/Booking/IBookingService.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+{
+    public interface IBookingService
```
Comment (CrazyKamenwik) [inline BKG/Booking.BLL/Services/Booking/IBookingService.cs]: It's better to inherit service's interface from base or generic service's interface for enhance it's functionality.

File: BKG/Booking.BLL/Services/Booking/BookingService.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+using Booking.DAL.Repositories.Booking;
+using Booking.DAL.Repositories.Generic;
+
+namespace Booking.BLL.Services.Booking
+{
+    public class BookingService : GenericService<BookingEntity, BookingModel>, IBookingService
+    {
+        private readonly IBookingRepository _bookingRepository;
+
+        public BookingService(IGenericRepository<BookingEntity> repository,
```
Comment (CrazyKamenwik) [inline BKG/Booking.BLL/Services/Booking/BookingService.cs]: If you inherit IBookingRepository from IGenericRepository you will avoid this redundant passing of IGenericRepository into base class

**Review by CrazyKamenwik — APPROVED** [review]

**Review by nantonov — APPROVED** [review]

**Pull request merged**